bitcoin/src/sign_message.rs

```diff
@@ -151,7 +151,7 @@ mod message_signing {
             match address.address_type() {
                 Some(AddressType::P2pkh) => {
                     let pubkey = self.recover_pubkey(secp_ctx, msg_hash)?;
-                    Ok(*address == Address::p2pkh(&pubkey, *address.network()))
+                    Ok(address.pubkey_hash().unwrap() == pubkey.pubkey_hash())
```

Comment:
In 54b80450c5778744f05ad63bd8df1c69b9c807e1:

This `unwrap` could be replaced by wrapping the other item in `Some`.

In general I'm a little disappointed that you now need to unwrap so many script constructors. But I guess there's not really an alternative and it could be really bad if we allow invalid/unspendable scripts to be created.

Reply:
I'm surprised that I wrote `unwrap` at all, I haven't done that for a long time but this PR is pretty old. I'll check over all the unwraps and see if I can do something better. Thanks for noticinng.